src/transformers/models/audioflamingo3/modular_audioflamingo3.py

```diff
@@ -0,0 +1,457 @@
+    """
+
+    def __init__(self, config: AudioFlamingo3Config):
+        super().__init__()
+        self.linear_1 = nn.Linear(
+            config.audio_config.intermediate_size, config.text_config.hidden_size, bias=config.projector_bias
+        )
+        self.act = ACT2FN[config.projector_hidden_act]
+        self.linear_2 = nn.Linear(
+            config.text_config.hidden_size, config.text_config.hidden_size, bias=config.projector_bias
+        )
+
+    def forward(self, audio_features):
+        hidden_states = self.linear_1(audio_features)
+        hidden_states = self.act(hidden_states)
+        hidden_states = self.linear_2(hidden_states)
+        return hidden_states
+
+
+@auto_docstring(
+    custom_intro="""
+    The AudioFlamingo3 model which consists of a audio backbone and a language model.
+    """
+)
+class AudioFlamingo3ForConditionalGeneration(AudioFlamingo3PreTrainedModel, GenerationMixin):
+    """
+    AudioFlamingo3 model composed of an audio encoder, a projection to the LM hidden size, and a causal LM.
+
+    The audio-text fusion is performed by *replacing* occurrences of the `<sound>` token with per-frame audio embeddings,
+    without changing sequence length. The number of `<sound>` tokens is expected to match the *post-pool* frame count
+    computed by the processor.
+    """
+
+    config_class = AudioFlamingo3Config
+
+    def __init__(self, config: AudioFlamingo3Config):
+        super().__init__(config)
+        # Language model
+        self.language_model = AutoModelForCausalLM.from_config(config.text_config)
+        # Audio encoder (explicitly instantiate our class to guarantee helper availability)
+        self.audio_tower = AudioFlamingo3Encoder(config.audio_config)
+        # Projection to LM hidden size
+        self.multi_modal_projector = AudioFlamingo3MultiModalProjector(config)
+
+        self.post_init()
+
+    # --- Embedding plumbing (forward to LM) ---
+    def get_input_embeddings(self):
+        return self.language_model.get_input_embeddings()
+
+    def set_input_embeddings(self, value):
+        self.language_model.set_input_embeddings(value)
+
+    def get_output_embeddings(self):
+        return self.language_model.get_output_embeddings()
+
+    def set_output_embeddings(self, value):
+        self.language_model.set_output_embeddings(value)
+
+    def set_decoder(self, decoder):
+        self.language_model.set_decoder(decoder)
+
+    def get_decoder(self):
+        return self.language_model.get_decoder()
+
+    @can_return_tuple
+    @auto_docstring
+    def forward(
+        self,
+        input_ids: Optional[torch.LongTensor] = None,
+        input_features: Optional[torch.FloatTensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        feature_attention_mask: Optional[torch.Tensor] = None,
+        # TODO (ebezzam) why unused? normally passed to self.language_model
+        position_ids: Optional[torch.LongTensor] = None,
```

Comment:
Is there a reason to not pass to self.language model? Like in Qwen below

https://github.com/huggingface/transformers/blob/a8cece13e29057fc9712543402d73e78796b306d/src/transformers/models/qwen2_audio/modeling_qwen2_audio.py#L821